## Exploración de los datos

### By:
Miguel Angel Toro Ramos

### Date:
2026-08-20

### Description:

el objetivo es realizar una exploración general de datos para verificar los tipos de datos con el fin de comprender de las características y el esquema de datos y si es posible solucionar problemas básicos relacionados.

## 📚 Import  libraries

In [1]:
# base libraries for data science
from pathlib import Path

import pandas as pd
import pyarrow as pa

## 💾 Load data

In [2]:
# data directory path
DATA_DIR = Path.cwd().resolve().parents[0] / "Admisiones-predict-project" / "data"

print(f"Data directory path: {DATA_DIR}")

admisiones_df = pd.read_csv(DATA_DIR / "01_raw/Admission_Predict.csv")

Data directory path: /home/dell/Admisiones-predict-project/data


## 📊 Data description

In [3]:
admisiones_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 623 entries, 0 to 622
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   GRE Score          612 non-null    float64
 1   TOEFL Score        604 non-null    float64
 2   University Rating  617 non-null    float64
 3   SOP                607 non-null    float64
 4   LOR                615 non-null    float64
 5   CGPA               621 non-null    float64
 6   Research           594 non-null    float64
 7   Chance of Admit    623 non-null    float64
dtypes: float64(8)
memory usage: 39.1 KB


In [4]:
admisiones_df.sample(20)

,GRE Score,TOEFL Score,University Rating,SOP,LOR,CGPA,Research,Chance of Admit
73,314.0,108.0,4.0,4.5,4.0,9.04,1.0,0.84
469,332.0,118.0,2.0,4.5,3.5,9.36,1.0,0.90
102,314.0,106.0,2.0,4.0,3.5,8.25,0.0,0.62
531,317.0,103.0,3.0,2.5,3.0,8.54,1.0,0.73
274,315.0,100.0,1.0,2.0,2.5,7.95,0.0,0.58
214,331.0,117.0,4.0,4.5,5.0,9.42,1.0,0.94
64,325.0,111.0,3.0,3.0,3.5,8.70,0.0,0.52
197,310.0,106.0,2.0,3.5,2.5,8.33,0.0,0.73
358,314.0,105.0,2.0,2.5,2.0,7.64,0.0,0.70
335,325.0,111.0,4.0,4.0,4.5,9.11,1.0,0.83


## Revision estadistica inicial de los datos

In [5]:
admisiones_df.describe()

,GRE Score,TOEFL Score,University Rating,SOP,LOR,CGPA,Research,Chance of Admit
count,612.000000,604.000000,617.000000,607.000000,615.000000,621.000000,594.000000,623.000000
mean,317.094771,107.865894,3.178282,3.456343,3.517886,8.637681,0.550505,0.733034
std,11.747793,6.116630,1.154037,1.005015,0.867719,0.589548,0.497862,0.142240
min,290.000000,92.000000,1.000000,1.000000,1.000000,6.800000,0.000000,0.340000
25%,309.000000,104.000000,2.000000,3.000000,3.000000,8.220000,0.000000,0.640000
50%,317.000000,108.000000,3.000000,3.500000,3.500000,8.650000,1.000000,0.730000
75%,326.000000,112.000000,4.000000,4.000000,4.000000,9.100000,1.000000,0.840000
max,340.000000,120.000000,5.000000,5.000000,5.000000,9.920000,1.000000,0.970000


## Convertir tipos de datos

In [6]:
cols_integer = ["GRE Score", "TOEFL Score", "University Rating"]
admisiones_df[cols_integer] = admisiones_df[cols_integer].astype("Int64")

In [7]:
admisiones_df["Research"] = (
    pd.to_numeric(admisiones_df["Research"], errors="coerce")
    .map({0.0: False, 1.0: True})
    .astype("boolean")
)

In [8]:
admisiones_df["Research"].value_counts(dropna=False)

Research
True     327
False    267
<NA>      29
Name: count, dtype: Int64

In [9]:
admisiones_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 623 entries, 0 to 622
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   GRE Score          612 non-null    Int64  
 1   TOEFL Score        604 non-null    Int64  
 2   University Rating  617 non-null    Int64  
 3   SOP                607 non-null    float64
 4   LOR                615 non-null    float64
 5   CGPA               621 non-null    float64
 6   Research           594 non-null    boolean
 7   Chance of Admit    623 non-null    float64
dtypes: Int64(3), boolean(1), float64(4)
memory usage: 37.2 KB


In [10]:
schema = pa.Table.from_pandas(admisiones_df).schema

## 💾 Save dataframe with data types

In [11]:
processed_path = DATA_DIR / "02_intermediate/admisiones_type_fixed.parquet"
admisiones_df.to_parquet(
    processed_path,
    index=False,
    schema=schema,
)
pd.read_parquet(processed_path).info()

<class 'pandas.DataFrame'>
RangeIndex: 623 entries, 0 to 622
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   GRE Score          612 non-null    Int64  
 1   TOEFL Score        604 non-null    Int64  
 2   University Rating  617 non-null    Int64  
 3   SOP                607 non-null    float64
 4   LOR                615 non-null    float64
 5   CGPA               621 non-null    float64
 6   Research           594 non-null    boolean
 7   Chance of Admit    623 non-null    float64
dtypes: Int64(3), boolean(1), float64(4)
memory usage: 37.2 KB


## 📊 Analysis of Results and Conclusions 

### Resultados y conclusiones

El conjunto de datos contiene 623 registros y 8 variables. La variable objetivo,
`Chance of Admit`, no presenta valores faltantes y tiene un promedio de 0.73,
con valores entre 0.34 y 0.97.

Se identificaron valores faltantes en las variables predictoras. `Research`
presenta la mayor proporción de datos faltantes (29 registros, 4.65 %), seguida
por `TOEFL Score` (19 registros, 3.05 %) y `SOP` (16 registros, 2.57 %).
Las demás variables presentan menos del 2 % de valores faltantes. Los valores
faltantes se conservaron como valores nulos, evitando reemplazarlos por cero o
por estadísticas sin una justificación basada en el contexto o en el modelo.

Se corrigieron los tipos de datos para representar adecuadamente el esquema:
`GRE Score`, `TOEFL Score` y `University Rating` se almacenaron como enteros
anulables; `Research` como booleano anulable; y las variables `SOP`, `LOR`,
`CGPA` y `Chance of Admit` se conservaron como numéricas decimales.

Finalmente, el dataset procesado se almacena en formato Parquet, que preserva
el esquema y los valores faltantes. Para la etapa de modelado se deberá definir
una estrategia de imputación usando únicamente los datos de entrenamiento,
con el fin de evitar fuga de información.
